pkg/rcache/rcache.go

```
@@ -57,105 +57,181 @@ func redisPool() (*pool.Pool, error) {
 	return connPool_, nil
 }
 
-// Redis is a cache implemented on top of a Redis client. It is
-// designed to mimick the API of cache.Cache to make it easy to switch
-// instances of cache.Cache to Redis.
+// getConn returns a redis client from the pool. When you are done you must
+// call the cleanup function to return the connection to the pool.
+func getConn() (*redis.Client, func(), error) {
+	connPool, err := redisPool()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	conn, err := connPool.Get()
+	if err != nil {
+		return nil, nil, err
+	}
+	return conn, func() { connPool.Put(conn) }, nil
+}
+
+// Redis exposes methods for speaking to the configured redis server. It
+// serves two main functions: Ensure we namespace keys and correct usage of
+// the connection pool.
 type Redis struct {
```

The contract of `Cache` is different. We silently ignore errors in `ByteCache` and treat it as a cache miss. In `Cache` we do not.

> I don't believe that tech debt with be tackled unless we do it while solving real issues.

Agreed, but at the same time I am tired of opsgenie waking me up at 4am. How about I promise to you I'll follow up straight away on this (ie right after merge), I just want to get this out into production to see if it works.